modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQJMSContext.java

```diff
@@ -0,0 +1,729 @@
+            if (logger.isDebugEnabled()) {
+                logger.debug("Created Browser for: {} with Selector: {}",
+                        new Object[]{queue, messageSelector});
+            }
+            return queueBrowser;
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public TemporaryQueue createTemporaryQueue() {
+        try {
+            TemporaryQueue temporaryQueue = getSession().createTemporaryQueue();
+            logger.debug("Created Temporary Queue");
+            return temporaryQueue;
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public TemporaryTopic createTemporaryTopic() {
+        try {
+            TemporaryTopic temporaryTopic = getSession().createTemporaryTopic();
+            logger.debug("Created Temporary Topic");
+            return temporaryTopic;
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     * TODO: Should throw InvalidDestinationRuntimeException if invalid name.
+     *
+     * @param name name used to identify this subscription
+     */
+    @Override
+    public void unsubscribe(String name) {
+        try {
+            getSession().unsubscribe(name);
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void acknowledge() {
+        try {
+            getSession().acknowledge();
+        } catch (IllegalStateException e) {
+            throw new IllegalStateRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * Method to retrieve session.
+     * Creates session if not already created.
+     *
+     * @return session currently applicable session
+     */
+    private AMQSession getSession() {
+        if (session == null) {
+            try {
+                session = (AMQSession) connection.createSession(getSessionMode());
+            } catch (JMSException e) {
+                throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+            }
+        }
+        return session;
+    }
+
+    /**
+     * Method to retrieve the single message producer of the JMSContext.
+     * Creates message producer if not already created.
+     *
+     * @return MessageProducer the shared MessageProducer
+     */
+    private MessageProducer getSharedMessageProducer() {
+        if (sharedMessageProducer == null) {
+            try {
+                sharedMessageProducer = session.createProducer(null);
+            } catch (java.lang.IllegalStateException e) {
```

Comment:
The createProducer() call on a Session in andes throws java.lang.IllegalStateException while some other methods such as acknowledge() throw javax.jms.IllegalStateException. Hence the inclusion of the fully qualified class path here.